src/main/groovy/life/qbic/portal/sampletracking/ui/SampleImport.groovy

```diff
@@ -12,29 +17,133 @@ class SampleImport extends VerticalLayout {
+            // Get value from user input in textField
+            String sampleIdInput = this.additionalSampleId.getValue()
+            try {
+                this.controller.querySampleById(sampleIdInput)
+                // if sample was found show success notification
+                Notification notification = new Notification
+                        (
+                                "Success", "Sample Id: $sampleIdInput was added to Sample List",
+                                Notification.Type.HUMANIZED_MESSAGE)
+                notification.setDelayMsec(3000)
+                notification.setPosition(Position.TOP_CENTER)
+                notification.show(Page.getCurrent())
+
+                //ToDo style notification e.g.
+                // make notification background color for clearer distinction between
+                // success and failed sample registration attempt or change Sample Id to be bold and italic
+
+            }
+            catch (Exception e) {
+                // show notification if user input a sample id but it couldn't be found
+                if (sampleIdInput) {
+                    Notification notification = new Notification
+                            (
+                                    "Sample Id Error", "Sample Id: $sampleIdInput could not be found in database",
+                                    Notification.Type.ERROR_MESSAGE)
+                    notification.setDelayMsec(3000)
+                    notification.setPosition(Position.TOP_CENTER)
+                    notification.show(Page.getCurrent())
+
+                }
+                //show notification if no sample id was provided
+                else {
+                    Notification notification = new Notification
+                            (
+                                    "No Sample Id Input", "Please Input a Sample Id",
+                                    Notification.Type.ERROR_MESSAGE)
+                    notification.setDelayMsec(3000)
+                    notification.setPosition(Position.TOP_CENTER)
+                    notification.show(Page.getCurrent())
+
+                }
+
+            }
+        })
+        // display notification that file upload was successful and samples were added to grid
+        this.fileSampleAddUpload.addSucceededListener({ event ->
+
+            Upload.Receiver receiver = this.fileSampleAddUpload.getReceiver()
+            FileOutputStream uploadedFileStream = receiver.receiveUpload("temp", ".csv")
+            // display notification if file upload and addition to sample list was successful
+
+            //ToDo check if uploadedfileStream is empty
+            // to avoid success notification to user when no file was selected
+
+
+            if (uploadedFileStream) {
+
+                //ToDo implement controller access of uploaded file here,
+                // should an outputStream be provided as File or should the samples be extracted here into List of Samples?
```

Comment:
I would provide a `OutputStream`. The extraction should not be handled by the view here in my opinion.